src/python/tests/test_maf_to_fasta.py

```diff
@@ -0,0 +1,131 @@
+        [
+            ('gabs.maf', 'gabs', 'gab_genomes.txt', ['0.fa', '0.json', '1.fa', '1.json'], does_not_raise()),
+            ('gabs.maf', 'gabs', 'wrong_genomes.txt', [], raises(ValueError)),
+            ('gabs.maf', 'gabs', None, [], raises(ValueError))
+        ]
+    )
+    def test_convert_maf_to_fasta(self, maf_file: PathLike, output_dir: PathLike,
+                                  genomes_file: Optional[PathLike], out_file_rel_paths: List[str],
+                                  expectation: ContextManager, tmp_dir: Path) -> None:
+        """Tests :func:`maf_to_fasta.convert_maf_to_fasta()` function.
+
+        Args:
+            maf_file: Input MAF file with alignment blocks. The src fields of
+                this MAF file should be of the form '<genome>.<seqid>'.
+            output_dir: Output directory under which FASTA and JSON files will be created.
+            genomes_file: File listing the genomes in the input MAF file, one per line.
+            out_file_rel_paths: Relative paths of the output FASTA and JSON files.
+            expectation: Context manager for the expected exception. The test will only pass if that
+                exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+            tmp_dir: Unit test temp directory (fixture).
+
+        """
+        maf_file_path = self.ref_file_dir / maf_file
+        out_dir_path = tmp_dir / output_dir
+        if genomes_file is not None:
+            genomes_file = self.ref_file_dir / genomes_file
+
+        with expectation:
+            maf_to_fasta.convert_maf_to_fasta(maf_file_path, out_dir_path, genomes_file=genomes_file)
+
+        ref_dir_path = self.ref_file_dir / output_dir
+        for out_file_rel_path in out_file_rel_paths:
+            out_file_path = out_dir_path / out_file_rel_path
+            ref_file_path = ref_dir_path / out_file_rel_path
+            assert filecmp.cmp(out_file_path, ref_file_path)
+
+    @pytest.mark.parametrize(
+        "non_negative_integer, exp_output, expectation",
+        [
+            (0, Path('.'), does_not_raise()),
+            (10, Path('0'), does_not_raise()),
+            ('75', None, raises(TypeError)),
+            (-1, None, raises(ValueError))
+        ]
+    )
+    def test_map_uint_to_path(self, non_negative_integer: int, exp_output: Path,
+                              expectation: ContextManager) -> None:
+        """Tests :func:`maf_to_fasta.map_uint_to_path()` function.
+
+        Args:
+            non_negative_integer: A non-negative integer.
+            exp_output: Expected return value of the function.
+            expectation: Context manager for the expected exception. The test will only pass if that
+                exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+
+        """
+        with expectation:
+            obs_output = maf_to_fasta.map_uint_to_path(non_negative_integer)
+            assert obs_output == exp_output
```

Comment:
I would suggest including more tests regarding if files don't exist, files containing wrong maf, etc.

Reply:
The `map_uint_to_path` function restricts itself to creating `pathlib.Path` objects without making any changes to the file system. Unit testing of whether files exist or contain invalid MAF data might arguably be written for `convert_maf_to_fasta`, but file handling is already done by base Python, and MAF processing is already done by the [Biopython MafIterator](https://github.com/biopython/biopython/blob/a395b5c3832161d7387eabaa4bc2e60b2d12e5c1/Bio/AlignIO/MafIO.py#L129-L250). I would be reluctant to reinvent the wheel in this case.